$OpenBSD: patch-psi_dxmain_c,v 1.2 2017/10/17 13:22:47 stsp Exp $

Fix a stack smash.
Similar to upstream git commit 5192635686b3a9da14e4531c7ce7ca85cd629a66

Index: psi/dxmain.c
--- psi/dxmain.c.orig
+++ psi/dxmain.c
@@ -82,7 +82,9 @@ read_stdin_handler(GIOChannel *channel, GIOCondition c
         input->count = 0;	/* EOF */
     }
     else if (condition & (G_IO_IN)) {
-        g_io_channel_read_chars(channel, input->buf, input->len, (gsize *)&input->count, &error);
+        gsize gsz;
+        g_io_channel_read_chars(channel, input->buf, input->len, &gsz, &error);
+        input->count = (int)gsz;
         if (error) {
             g_print("%s\n", error->message);
             g_error_free(error);
